plugin/ensime.vim

```diff
@@ -38,6 +37,10 @@ command! -nargs=* -range EnToggleFullType call ensime#com_en_toggle_fulltype([<f
 command! -nargs=* -range EnOrganizeImports call ensime#com_en_organize_imports([<f-args>], '')
 command! -nargs=* -range EnAddImport call ensime#com_en_add_import([<f-args>], '')
 
+function! EnPackageDecl() abort
+  return ensime#com_en_package_decl()
+endfunction
```

Comment:
`@execute_with_client()` was introduced explicitly to remove the incredible boilerplate code that was available before and `(args, range)` allows us to generalize over all the methods and gives us the ability to customize the external API of a method in an easy way. AFAIK, it is a popular pattern used throughout all the vim plugins and community. It was present in the plugin even before I started to contribute.
  
Regarding the issue that you point out about `autoload` and the wrappers, could you please elaborate on what you mean and what are you proposing? Good ideas are welcome :+1: 
  
And yes, `com_` comes from 'command'.